### PR TITLE
Bump to Pytest 6.2.2 with -p no:unraisableexception -p no:threadexception

### PR DESCRIPTION
**Pull request by altendky**
See #1878 for the initial dependabot PR showing the build issues.

https://docs.pytest.org/en/stable/changelog.html (See 6.2.0 features)
https://docs.pytest.org/en/stable/usage.html#unraisable

I think that disabling these new plugins for now is reasonable since it allows us to keep moving forward with pytest versions while retaining the same lack of these checks as if we held the version back.  If there is concern about globally disabling these checks then I think a follow up ticket and PR would be appropriate.

I did make a cursory effort to use `@pytest.mark.filterwarnings()` but failed to get it to successfully avoid the new errors.  I suspect something related to the combination with `-W error` and other pytest features but I did not dig into it.

### CODE REVIEW

**Comment by sbraz** [comment]
Hi @altendky, I have never used dependabot before so I wouldn't know how to help. I have not looked too hard at this issue, I just found the failure with newer pytest versions quite weird.

As for #1868, it did have the one failure that the pytest bump created (but the rest was unrelated, I agree):
```
FAILED trio/_core/tests/test_asyncgen.py::test_fallback_when_no_hook_claims_it - AssertionError: assert 'ignored GeneratorExit' in ''
```

File: setup.cfg [hunk 1]
```diff
@@ -4,3 +4,6 @@ faulthandler_timeout=60
 markers =
     redistributors_should_skip: tests that should be skipped by downstream redistributors
 junit_family = xunit2
+addopts =
```
Comment (graingert) [inline setup.cfg:7]: @altendky if you set

```
filterwarnings =
    error
    ignore::pytest.PytestUnraisableExceptionWarning
    ignore::pytest.PytestUnhandledThreadExceptionWarning
```

and remove the `-W error` does that work?

**Comment by graingert** [comment]
@altendky I've had some success running the suite with stepwise and wrapping the places that currently raise in:


```python
@contextlib.contextmanager
def restore_unraisable_hook():
    sys.unraisablehook, prev = sys.__unraisablehook__, sys.unraisablehook
    try:
        yield
    finally:
        sys.unraisablehook = prev
```

```python
    with restore_unraisable_hook(), pytest.warns(RuntimeWarning, match="Trio guest run got abandoned"):
        do_abandoned_guest_run()
        gc_collect_harder()
```

**Comment by graingert** [comment]
eg some of the code is relying on the default unraisablehook printing to stderr:

https://github.com/python-trio/trio/blob/58e9b80c283caf5dd95a9bb1aaf12e49a47e9bcf/trio/_core/tests
/test_asyncgen.py#L270-L281

**Review by graingert — APPROVED** [review]

**Pull request merged**